llama_stack/providers/utils/inference/model_registry.py

```diff
@@ -60,8 +61,33 @@ def get_llama_model(self, provider_model_id: str) -> str:
             return None
 
     async def register_model(self, model: Model) -> Model:
-        model.provider_resource_id = self.get_provider_model_id(
-            model.provider_resource_id
-        )
+        provider_resource_id = self.get_provider_model_id(model.provider_resource_id)
+        if provider_resource_id:
+            model.provider_resource_id = provider_resource_id
+        else:
+            if model.metadata.get("llama_model") is None:
+                raise ValueError(
+                    f"Model '{model.provider_resource_id}' is not available and no llama_model was specified in metadata. "
+                    "Please specify a llama_model in metadata or use a supported model identifier"
+                )
+            existing_llama_model = self.get_llama_model(model.provider_resource_id)
+            if existing_llama_model:
+                if existing_llama_model != model.metadata["llama_model"]:
+                    raise ValueError(
+                        f"Provider model id '{model.provider_resource_id}' is already registered to a different llama model: '{existing_llama_model}'"
+                    )
+            else:
+                # Validate that the llama model is a valid CoreModelId
```

Comment:
this is not right or at least not future proof. I think you should only accept huggingface names as values for llama model. so collect a list of all huggingface repos allowed and check against that

Reply:
done.